---
title: " Holobionte Simulation"
date: "`r Sys.Date()`"
params: 
  scenario: "microbiome"
  run: 1
output:
  html_document:
    code_folding: hide
    highlight: espresso
    theme: yeti
    toc: yes
    toc_depth: 3
    toc_float: yes
    
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

```

```{r library,  message=FALSE,fig.width=10,fig.height=6,warning=FALSE}

library(snpStats)
library(magrittr)
library(data.table)
library(tidyverse)
library(devtools)
library(FactoMineR)
library(utils)
library(BGLR)
library(bigsnpr)
library(knitr)
library(rmarkdown)
library(tictoc)
 theme_set( theme_classic())



```

```{r}
scenario <- params$scenario
run <- params$run
```

[//]: # "Load the various chunks in the various files"
```{r readChunks,  warning=FALSE, message=FALSE}
WORKING_DIR   = "~/work/holobionte_wp2/code/"

#PATH ##########################################################################

path_pheno = "../data/Vanille_Deru/donnees_pheno/"
path_OTU = "../data/Vanille_Deru/donnees_microbiote/"
path_output = "../data/Vanille_Deru/simulation/" 
path_geno =  "../data/Vanille_Deru/donnees_geno/"
path_RDS = paste0("../data/RDS/",scenario,"/",as.character(run),"/")
path_simu = path_output

# folder containing snp and otu data
data1 = "../data/Pérez/" #PEREZ
data2 = "../data/Vanille_Deru/simulation/" # VANILLE DERU


 ### Define working folder (contains R/Rmd file for current sample, parent contains global project files)

analysisParamsFilePath = file.path( WORKING_DIR, "Params.R")


paramsEnv = new.env();
source(analysisParamsFilePath, local = paramsEnv);

source( file.path( WORKING_DIR, "Load_parameter.R"));
source("Params.R")
source("Simubiome.R")


read_chunk( path=file.path( WORKING_DIR,"Compte.R"))
read_chunk( path=file.path( WORKING_DIR,"Simuscript.R"))
read_chunk( path=file.path( WORKING_DIR,"Vanille_Préparation_données.R"))
read_chunk( path=file.path( WORKING_DIR,"PCA-Deru.R"))
read_chunk( path=file.path( WORKING_DIR,"Cluster.R"))
read_chunk( path=file.path( WORKING_DIR,"Cluster_B.R"))

```



### Environment  {.tabset}

#### Parameters 

<center>
```{r runInfo,  message=FALSE}

showSimpleDT( paramsEnv)
```



#### R

```{r runInfo_r,  message=FALSE}
print( sessionInfo());

```

### simulation

``` {r simu,  warning=FALSE, message=FALSE,warning=FALSE,fig.width=15,fig.height=15, results='asis'} 
<<simu>>
```



### Réduction de dimension {.tabset}

#### PCA Analysis Genome

```{r PCA_Deru,  warning=FALSE, message=FALSE,warning=FALSE,fig.width=15,fig.height=15, results='asis'}
<<PCA_Deru>>
```


#### Clusterisation génome

```{r Cluster_X,  warning=FALSE, message=FALSE,warning=FALSE,fig.width=15,fig.height=15, results='asis'}
<<Cluster_X>>
```

#### analyses quantitatives génome

```{r quantitative_X,  warning=FALSE, message=FALSE,warning=FALSE,fig.width=15,fig.height=15, results='asis'}
<<quantitative_X>>
```

#### PCA Analysis Microbiote

```{r PCA_micro,  warning=FALSE, message=FALSE,warning=FALSE,fig.width=15,fig.height=15, results='asis'}
<<PCA_micro>>
```

#### Clusterisation microbiote

```{r Cluster_B,  warning=FALSE, message=FALSE,warning=FALSE,fig.width=15,fig.height=15, results='asis'}
<<Cluster_B>>
```



#### analyses quantitatives microbiote

```{r quantitative_B,  warning=FALSE, message=FALSE,warning=FALSE,fig.width=15,fig.height=15, results='asis'}
<<quantitative_B>>
```




